Sources/Utility/ArgumentParser.swift

```diff
@@ -265,8 +266,39 @@ public final class ArgumentParser: ArgumentParserProtocol {
         public func get<T>(_ arg: PositionalArgument<T>) -> T {
             return results[AnyArgument(arg)] as! T
         }
+
+        /// Get the subparser which was chosen.
+        public func get(_ subparser: Subparser) -> String {
+            return subparser.result!
+        }
+
+        /// Merge these results with another result.
+        fileprivate mutating func merge(_ result: Result) {
+            for (k, v) in result.results {
+                results[k] = v
+            }
+        }
+    }
+
+    /// This class holds additional argument parsers
+    /// used in a sub-tool style model.
+    public class Subparser {
+        /// The mapping of parsers to their subcommand.
+        private(set) var parsers: [String: ArgumentParser] = [:]
+
+        /// The subcommand that was chosen after command line parsing.
+        fileprivate var result: String?
+
+        /// Add a parser with a subcommand name and its corresponding overview.
+        func add(parser command: String, overview: String) -> ArgumentParser {
```

Comment:
Instead of passing `add()` the parameters and getting back a parser, it would be better to be able to allow a chained parser to be instantiated in the normal way and then just added to a parser.  This would allow sharing of chained parsers, where that makes sense.